test/functional/helpers.lua

```diff
@@ -508,7 +508,9 @@ end
 
 -- PowerShell 5.1; comes with and available only on Windows
 function module.has_powershell()
-  return module.eval('executable("powershell")') == 1
+  -- Ensure we're on Windows in case, for whatever reason, pwsh is
+  -- aliased to powershell like it apparently is on macos test runners.
```

Comment:
Come up with a different os-agnostic check for `has_powershell` (and `has_pwsh`) to tolerate aliases or revert the change in `has_powershell`.

Reply:
Sorry, I don't understand what you mean. Originally, both functions literally had the same `iswin()` check in them, I did not change anything there. The only thing that changed is has_powershell() guarded tests now only run on Windows.

Test suite assumed `pwsh` and `powershell` commands can be used interchangeably, and they can't (at least not at all times). `powershell` is PowerShell 5 and exists only on Windows and `pwsh` is PowerShell Core which exhibits different behaviour and allow us to use a different recommended configuration, which sometimes produces a different output, and thus needs a different set of test reference values.

So, test writers are basically dealing with two different shells here that may need to be tested separately. 

The split also allows us to now test both PowerShell 5 and PowerShell Core on Windows runners (previously, only PowerShell 5 was tested)

Reply:
PTAL. I'm not sure I understood you correctly, but I reverted the change and right now we have

* has_powershell() -- tries windows powershell 5 on windows and falls back to powershell core everywhere else
* has_pwsh() -- specifically checks for powershell core